Source/Core/DolphinQt/HotkeyScheduler.cpp

```diff
@@ -143,8 +145,9 @@ void HotkeyScheduler::Run()
 
     if (Core::GetState() != Core::State::Stopping)
     {
-      // Obey window focus before checking hotkeys.
-      Core::UpdateInputGate();
+      // Obey window focus (config permitting) before checking hotkeys.
+      ControlReference::SetInputGate(Host_RendererHasFocus() ||
+                                     !Config::Get(Config::MAIN_FOCUSED_HOTKEYS));
```

Comment:
Doesn't this also need `&& !Host_UIBlocksControllerState()` like `UpdateInputGate`, otherwise weird behavior will happen?

Reply:
Host_UIBlocksControllerState is only used when imgui grabs input if I remember correctly. I think hotkeys should still function in that situation.

Reply:
Ah, well I think it still needs to be checked, otherwise hotkeys will activate while typing into an ImGui text input. We don't currently have any of those, but we may eventually.

Reply:
Fixed.